game/src/LocaleButton.gd

```diff
@@ -21,15 +24,32 @@ func _ready():
 	Events.Options.load_settings.connect(load_setting)
 	Events.Options.save_settings.connect(save_setting)
 
-
-func load_setting(file : ConfigFile):
-	var locale_index := _locales_list.find(file.get_value("Localization", "Locale", "") as String)
-	if locale_index != -1:
-		selected = locale_index
-
-func save_setting(file : ConfigFile):
-	file.set_value("Localization", "Locale", _locales_list[selected])
-
-func _on_item_selected(index):
-	TranslationServer.set_locale(_locales_list[index])
-	Events.Options.save_settings_from_file.call_deferred()
+func _valid_index(index : int) -> bool:
+	return 0 <= index and index < _locales_list.size()
+
+func load_setting(file : ConfigFile) -> void:
+	if file == null: return
+	var load_value = file.get_value(section_name, setting_name, TranslationServer.get_locale())
+	match typeof(load_value):
+		TYPE_STRING, TYPE_STRING_NAME:
+			var locale_index := _locales_list.find(load_value as String)
+			if locale_index != -1:
+				selected = locale_index
+				return
+	push_error("Setting value '%s' invalid for setting [%s] %s" % [load_value, section_name, setting_name])
+	reset_setting()
+
+func save_setting(file : ConfigFile) -> void:
+	if file == null: return
+	file.set_value(section_name, setting_name, _locales_list[selected])
+
+func reset_setting() -> void:
+	selected = _locales_list.find(TranslationServer.get_locale())
+
+func _on_item_selected(index : int) -> void:
+	if _valid_index(index):
+		TranslationServer.set_locale(_locales_list[index])
+		Events.Options.save_settings_to_file.call_deferred()
```

Comment:
One thing that is strange with this is that it will perform a double save if the player is in the options menu, the effects of this won't cause any issues right now, it'll only ever crop up to the player if we were to remove saving on application close and application crash (or in the case Godot is not running with a crash handler) but it is funky behavior in my opinion.